containerd-shim/process_linux.go

```diff
@@ -0,0 +1,96 @@
+// +build !solaris
+
+package main
+
+import (
+	"io"
+	"os"
+	"syscall"
+)
+
+// setPDeathSig sets the parent death signal to SIGKILL so that if the
+// shim dies the container process also dies.
+func setPDeathSig() *syscall.SysProcAttr {
+	return &syscall.SysProcAttr{
+		Pdeathsig: syscall.SIGKILL,
+	}
+}
+
+// openIO opens the pre-created fifo's for use with the container
+// in RDWR so that they remain open if the other side stops listening
+func (p *process) openIO() error {
+	p.stdio = &stdio{}
```

Comment:
Unless I read it incorrectly, this looks to be the exact same code that was in `process.go` initially, any reason why it was made a per arch function?